docs/preview/03-Features/01-Azure/01-service-bus.md

```diff
@@ -0,0 +1,279 @@
+    options.MaxConcurrentCalls = 5;
+
+    // Specifies the amount of messages that will be eagerly requested during processing.
+    // Setting the PrefetchCount to a value higher then the MaxConcurrentCalls value helps 
+    // maximizing throughput by allowing the message pump to receive from a local cache rather then waiting on a service request.
+    options.PrefetchCount = 10;
+
+    // The unique identifier for this background job to distinguish 
+    // this job instance in a multi-instance deployment (default: generated GUID).
+    options.JobId = Guid.NewGuid().ToString();
+
+    // Indicate whether or not the default built-in JSON deserialization should ignore additional members 
+    // when deserializing the incoming message (default: AdditionalMemberHandling.Error).
+    options.Routing.Deserialization.AdditionalMembers = AdditionalMemberHandling.Ignore;
+});
+```
+
+### Message handler routing customization
+The following routing options are available when registering an Azure Service Bus message handler on a message pump.
+
+> 💡 Setting one or more of the routing options helps the message pump to better select the right *message handler* for the received message.
+
+```csharp
+services.AddServiceBus[Topic/Queue]MessagePump(...)
+        .WithServiceBusMessageHandler<OrderMessageHandler, Order>(..., options =>
+        {
+            // Adds a filter to the message handler registration:
+            // Only messages with 'Type=Order' property goes through this message handler.
+            options.AddMessageContextFilter(ctx => ctx.Properties["Type"] == "Order");
+
+            // Adds a filter to the message handler registration:
+            // Only messages with certain bodies goes through this message handler.
+            options.AddMessageBodyFilter((Order order) => order.OrderId = "123");
+
+            // Adds a custom message deserializer to the message handler registration:
+            // Only messages that gets deserialized successfully goes through this message handler.
+            // 👉 Custom implementations of the `Arcus.Messaging.Abstractions.MessageHandling.IMessageBodySerializer` interface.
+            //      public interface IMessageBodySerializer
+            //      {
+            //          Task<MessageResult> DeserializeMessageAsync(string messageBody);
+            //      }
+            options.AddMessageBodySerializer(new CustomXmlMessageBodySerializer());
+        });
+```
+
+### Custom message settlement
+When messages can't be matched to any of your custom registered message handlers, Arcus will dead-letter the message. When one of your message handlers fails to process a message, it will get abandoned and maybe only after a third try be dead-lettered.
+
+This settlement of received Azure Service Bus messages can also be customized by calling one of the Service Bus operations yourself via the message context.
```

Comment:
Maybe we should mention here that it would be good practice for the application developer to deadletter the message himself when a non-transient/non retryable error occurs ?

Reply:
```suggestion
This settlement of received Azure Service Bus messages can also be customized by calling one of the Service Bus operations yourself via the message context.

> 🎖️ It is a good practice as application developer to dead-letter the message yourself when a non-transient/fatal error occurs.
```

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/546/commits/4da16920a98830b40b01cfac82f32c6ca7c38159